syntax = "proto3";

package crypto_condor;

// A single SPHINCS+ test vector.
message SphincsNistKatTest {
    int32 count = 1;
    string seed = 2;
    int32 mlen = 3;
    string msg = 4;
    string pk = 5;
    string sk = 6;
    int32 smlen = 7;
    string sm = 8;

    // Points to the start of the test vector inside the file.
    int32 line_number = 15;
}


// A group of test vectors for a single parameter set.
message SphincsNistKatVectors {
    // The name of the parameter set.
    string name = 1;
    repeated SphincsNistKatTest tests = 2;
}
